hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/HoodieRangeBucketIndex.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.index.bucket;
+
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecordLocation;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.table.HoodieTable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Range Bucket indexing mechanism.
+ */
+public class HoodieRangeBucketIndex extends HoodieBucketIndex {
+
+  private final int bucketRangeStepSize;
+
+  public HoodieRangeBucketIndex(HoodieWriteConfig config) {
+    super(config);
+    bucketRangeStepSize = config.getBucketRangeStepSize();
+  }
+
+  @Override
+  public boolean canIndexLogFiles() {
+    return false;
+  }
```

Comment:
I guess the log files can also be indexed here ?

Reply:
> I guess the log files can also be indexed here ?

may be indexed log files is not so useful in bucket scene. and in BaseSparkDeltaCommitActionExecutor.handleInsert, 
if set this true, the first insert will just write a log file but not baseFile, it's not a good chioce.
@YuweiXiao please help check this.

Reply:
@wqwl611 You are right, the first written will be log file only file group. I guess the overhead is ok as you need a compaction sooner or later when you have follow-up updates. 

By the way, if you set it to false, you will also produce log-file-only file group if you do not have your own partitioner.

Reply:
Just checked you new commits. Even you set true here, the first write should produce base file, right? As you guard the the empty file group writing.

Reply:
@YuweiXiao when init a big hudi table，produce log-file-only is not good chioce。
now I found the solution, please help check my new commits,  find is it proper？becase if not ‘super.handleInsert(fileId, recordItr);’， the sub handleInsert will write a log file when ‘table.getIndex().canIndexLogFiles()’ is true。

Reply:
I use a fixed uuid（00000000-0000-0000-0000-000000000000） to spell fileId, as shown below, do you think this is ok? please review my new commit。
@YuweiXiao 


<img width="979" alt="image" src="https://user-images.githubusercontent.com/67826098/193003627-e716d5cd-faa7-40dd-8961-8a77a7b20f12.png">